src/app/clusters/resource-monitoring-server/resource-monitoring-cluster-objects.h

```diff
@@ -141,37 +143,72 @@ struct TypeInfo
 };
 } // namespace LastChangedTime
 
-namespace ReplacementProductStruct {
-enum class Fields : uint8_t
+// A struct used during reads of the ReplacementProductList to store a single list instance we request
+// from the application.
+//
+// Inherit from an auto-generated struct to pick up the implementation bits, but make
+// it private inheritance so people can't accidentally use this struct where the other
+// is expected.
+struct GenericType : private HepaFilterMonitoring::Structs::ReplacementProductStruct::Type
 {
-    kProductIdentifierType  = 0,
-    kProductIdentifierValue = 1,
-};
+private:
+    char productIdentifierValueBuffer[kProductIdentifierValueMaxNameLength];
 
-struct Type
-{
 public:
-    ProductIdentifierTypeEnum productIdentifierType = static_cast<ProductIdentifierTypeEnum>(0);
-    chip::CharSpan productIdentifierValue;
-
-    CHIP_ERROR Decode(TLV::TLVReader & reader);
-
     static constexpr bool kIsFabricScoped = false;
-
-    CHIP_ERROR Encode(TLV::TLVWriter & writer, TLV::Tag tag) const;
+    virtual ~GenericType()                = default;
+    GenericType() {}
+    GenericType(ResourceMonitoring::ProductIdentifierTypeEnum aProductIdentifierType, chip::CharSpan aProductIdentifierValue)
+    {
+        setProductIdentifierType(aProductIdentifierType);
+        setProductIdentifierValue(aProductIdentifierValue);
+    }
+
+    using HepaFilterMonitoring::Structs::ReplacementProductStruct::Type::Encode;
+
+    /**
+     * Sets the product identifier type.
+     *
+     * @param aProductIdentifierType The product identifier type.
+     */
+    void setProductIdentifierType(ResourceMonitoring::ProductIdentifierTypeEnum aProductIdentifierType)
```

Comment:
```suggestion
    void SetProductIdentifierType(ResourceMonitoring::ProductIdentifierTypeEnum aProductIdentifierType)
```
and similar for other functions.